CMakeLists.txt

```diff
@@ -66,7 +70,13 @@ elseif(CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM64)
 elseif(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64)
   set(CLR_CMAKE_PLATFORM_ARCH_AMD64 1)
 elseif(WIN32)
-  set(CLR_CMAKE_PLATFORM_ARCH_AMD64 1)
+  if (CMAKE_GENERATOR MATCHES ".+ Win64")
```

Comment:
I would be against doing this as it means we couldn't use non VS generators. Thinking about it, could we pass the buildarch argument from the build script to CMake perhaps and use that? It could work for Linux too.